src/cryptonote_core/blockchain.cpp

```diff
@@ -2202,7 +2200,7 @@ bool Blockchain::get_blocks(uint64_t start_offset, size_t count, std::vector<std
 bool Blockchain::handle_get_objects(NOTIFY_REQUEST_GET_OBJECTS::request& arg, NOTIFY_RESPONSE_GET_OBJECTS::request& rsp)
```

Comment:
If we're going to use `RLOCK` here, I'd recommended marking this method as `const`.

Reply:
It's not just an optimization. It is to prevent further bugs in the code when someone tries to modify something inside a read-only lock.

Reply:
@SChernykh exactly. I am adding these (`const` qualifiers) to the final version of this PR.